### PR TITLE
Fix #2584. Remove flask_mail

### LINKED ISSUE #2584: replace flask.mail
flask.mail makes for an annoying coupling to flask.

### PR DESCRIPTION
**Pull request by e-carlin**
This is still a work in progress. I need to get postfix setup on my machine so I can test it.

### CODE REVIEW

File: sirepo/comsol_register.py [hunk 1]
```diff
@@ -6,51 +6,38 @@
...
-            MAIL_PORT=587,
-            MAIL_SERVER=cfg.mail_server,
-            MAIL_USERNAME=cfg.mail_username,
-            MAIL_PASSWORD=cfg.mail_password,
-        )
-        _mail = flask_mail.Mail(a)
-    req = http_request.parse_json()
-    msg = flask_mail.Message(
+    smtp.SMTP(
+        smtp_server=cfg.mail_server,
```
Comment (robnagler) [inline sirepo/comsol_register.py:27]: This should be moved into smtp. we don't need two smtp logins and moreover, we probably only want to send from one address/name (support) so add those with appropriate defaults to smtp cfg. recipient config needs to stay in comsol. 

I don't think we need to deprecate the config in auth.email. Just remove it. I can add it to the rsconf db in advance of pushing. Nobody else is using smtp (for comsol or auth).

File: sirepo/smtp.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        s = None
+        try:
+            # TODO(e-carlin): Understand line endings issue. EmailMessage
+            # defaults to \n RFC expects \r\n
+            # See https://docs.python.org/3/library/email.policy.html#email.policy.default
+            m = email.message.EmailMessage()
+            m['From'] = sender
+            m['To'] = recipient
+            m['Subject'] = subject
+            m.set_content = body
```
Comment (robnagler) [inline sirepo/smtp.py:31]: In [the documentation](https://docs.python.org/3/library/email.message.html#email.message.EmailMessage.set_content), it says this is function:

```py
set_content(*args, content_manager=None, **kw)
```

File: sirepo/smtp.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            with smtplib.SMTP(self.smtp_server, self.smtp_port) as s:
+                s.starttls()
+                s.ehlo()
+                s.login(self.smtp_user, self.smtp_password)
+                s.send_message(m)
+        finally:
```
Comment (robnagler) [inline sirepo/smtp.py:37]: why would this be necessary with the `with`?

File: sirepo/smtp.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        try:
+            # TODO(e-carlin): Understand line endings issue. EmailMessage
+            # defaults to \n RFC expects \r\n
+            # See https://docs.python.org/3/library/email.policy.html#email.policy.default
+            m = email.message.EmailMessage()
+            m['From'] = sender
+            m['To'] = recipient
+            m['Subject'] = subject
+            m.set_content = body
+            with smtplib.SMTP(self.smtp_server, self.smtp_port) as s:
```
Comment (robnagler) [inline sirepo/smtp.py:32]: I don't think you need the `try`. [`SMTP.__exit__`](https://github.com/python/cpython/blob/3.8/Lib/smtplib.py#L275) seems to do the right thing. [`quit`](https://github.com/python/cpython/blob/3.8/Lib/smtplib.py#L986) is designed for connection reuse, which we aren't doing.

File: sirepo/smtp.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    def __init__(*args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.smtp_port = 587
+
+    def send(self, subject, sender, recipient, body):
+        if isinstance(sender, tuple):
+            sender = f'{sender[0]} <{sender[1]}>'
+        s = None
+        try:
+            # TODO(e-carlin): Understand line endings issue. EmailMessage
```
Comment (robnagler) [inline sirepo/smtp.py:24]: I think [send always inserts `\r\n`](https://github.com/python/cpython/blob/3.8/Lib/smtplib.py#L968). `flatten` (in the referenced code) has to parse everything anyway.

File: sirepo/smtp.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+class SMTP(PKDict):
+    def __init__(*args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.smtp_port = 587
+
+    def send(self, subject, sender, recipient, body):
+        if isinstance(sender, tuple):
+            sender = f'{sender[0]} <{sender[1]}>'
+        s = None
```
Comment (robnagler) [inline sirepo/smtp.py:22]: remove

File: sirepo/comsol_register.py (hunk 1 above)
Comment (e-carlin) [reply sirepo/comsol_register.py]: https://github.com/radiasoft/sirepo/pull/2677/commits/5562b8e45c4790e870646d055a019adb1ceb5dc1

File: sirepo/smtp.py (hunk 2 above)
Comment (e-carlin) [reply sirepo/smtp.py]: Thanks https://github.com/radiasoft/sirepo/pull/2677/commits/5562b8e45c4790e870646d055a019adb1ceb5dc1

File: sirepo/smtp.py (hunk 4 above)
Comment (e-carlin) [reply sirepo/smtp.py]: Doh, I had the try initially, move to with, forgot to delete the try https://github.com/radiasoft/sirepo/pull/2677/commits/5562b8e45c4790e870646d055a019adb1ceb5dc1

File: sirepo/smtp.py (hunk 5 above)
Comment (e-carlin) [reply sirepo/smtp.py]: Nice, thx. https://github.com/radiasoft/sirepo/pull/2677/commits/5562b8e45c4790e870646d055a019adb1ceb5dc1

File: sirepo/smtp.py (hunk 6 above)
Comment (e-carlin) [reply sirepo/smtp.py]: https://github.com/radiasoft/sirepo/pull/2677/commits/5562b8e45c4790e870646d055a019adb1ceb5dc1

File: sirepo/smtp.py (hunk 3 above)
Comment (e-carlin) [reply sirepo/smtp.py]: https://github.com/radiasoft/sirepo/pull/2677/commits/5562b8e45c4790e870646d055a019adb1ceb5dc1

File: sirepo/smtp.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    m.set_content(body)
+    with smtplib.SMTP(cfg.server, cfg.port) as s:
+        s.starttls()
+        s.ehlo()
+        s.login(cfg.user, cfg.password)
+        s.send_message(m)
+
+
+cfg = pkconfig.init(
+    from_email=('support@sirepo.com', str, 'Email addres of sender'),
```
Comment (robnagler) [inline sirepo/smtp.py:30]: addres => address

Comment (e-carlin) [reply sirepo/smtp.py]: https://github.com/radiasoft/sirepo/pull/2677/commits/ddb33cd41cecd4a85c547fa764aa47e1f7fb0d03

**Pull request merged**